src/vmm/src/devices/virtio/block/block_metrics.rs

```diff
@@ -221,3 +221,181 @@ impl BlockDeviceMetrics {
             .add(other.io_engine_throttled_events.fetch_diff());
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+
+    #[test]
+    fn test_max_block_dev_metrics() {
+        // Note: this test has nothing to do with
+        // block structure or IRQs, this is just to allocate
+        // metrics for max number of devices that system can have.
+        // we have 5-23 irq for block devices on x86_64 so max
+        // 19 block devices and, although on aarch64 we have more
+        // devices but we stick to 19 to keep test common.
+        const MAX_BLOCK_DEVICES: usize = 19;
+
+        assert!(BLOCK_METRICS.read().is_ok());
+        assert!(BLOCK_METRICS.write().is_ok());
+
+        for i in 0..MAX_BLOCK_DEVICES {
+            let devn: String = format!("drv{}", i);
+            BlockMetricsPerDevice::alloc(devn.clone());
+            BLOCK_METRICS
```

Comment:
here or/and in the next loop, can we extract a common part somehow?

Reply:
done.

Reply:
what I mean is, it looks like there is quite a lot of repetition of long calls like `.read().unwrap().metrics.get().unwrap()`. I was wondering if it was possible to extract this to avoid repeating ourselves.

Reply:
ok, I changed it to how metrics is used through Block.